compiler/src/dotty/tools/dotc/interactive/Interactive.scala

```diff
@@ -269,8 +269,12 @@ object Interactive {
    *  @param includeReferences  If true, include references and not just definitions
    */
   def namedTrees(trees: List[SourceTree], includeReferences: Boolean, nameSubstring: String)
-   (implicit ctx: Context): List[SourceTree] =
-    namedTrees(trees, includeReferences, _.show.toString.contains(nameSubstring))
+   (implicit ctx: Context): List[SourceTree] = {
+    val predicate: NameTree => Boolean =
+      if (includeReferences) _.show.contains(nameSubstring)
```

Comment:
I think the code was broken before, but I don't understand why now we still have to print the whole tree in some situations. This does not match the documentation comment above, and I don't see why we would ever want to do this.

Reply:
I removed `includeReferences` from this function as it was never set. I have another fix that collects the names from the tree and check whether there exists one that contains the substring if you prefer to keep it.